skimage/_shared/tests/test_warnings.py

```diff
@@ -0,0 +1,166 @@
+        self._filename = filename
+        self.bad_filters = []
+        self.bad_stacklevels = []
+
+    def visit_Call(self, node):
+        p = ParseCall()
+        p.visit(node.func)
+        ast.NodeVisitor.generic_visit(self, node)
+
+        if str(self._filename) in file_whitelist:
+            return
+        if p.ls[-1] == 'simplefilter' or p.ls[-1] == 'filterwarnings':
+            if node.args[0].s == "ignore":
+                self.bad_filters.append(
+                    "{}:{}".format(self._filename, node.lineno))
+
+        if p.ls[-1] == 'warn' and (
+                len(p.ls) == 1 or p.ls[-2] == 'warnings'):
+
+            # See if stacklevel exists:
+            if len(node.args) == 3:
+                return
+            args = {kw.arg for kw in node.keywords}
+            if "stacklevel" not in args:
+                self.bad_stacklevels.append(
+                    "{}:{}".format(self._filename, node.lineno))
+
+
+@pytest.fixture(scope="session")
+def warning_calls():
+    # combined "ignore" and stacklevel error
+    base = Path(skimage.__file__).parent
+
+    bad_filters = []
+    bad_stacklevels = []
+
+    for path in base.rglob("*.py"):
+        # use tokenize to auto-detect encoding on systems where no
+        # default encoding is defined (e.g. LANG='C')
+        with tokenize.open(str(path)) as file:
+            tree = ast.parse(file.read(), filename=str(path))
+            finder = FindFuncs(path.relative_to(base))
+            finder.visit(tree)
+            bad_filters.extend(finder.bad_filters)
+            bad_stacklevels.extend(finder.bad_stacklevels)
+
+    return bad_filters, bad_stacklevels
+
+
+# @pytest.mark.slow
+def test_warning_calls_filters(warning_calls):
+    bad_filters, bad_stacklevels = warning_calls
+
+    if bad_filters:
+        raise AssertionError(
+            "warning ignore filter should not be in tests, instead, use\n"
+            "skimage._warnings.expected_warnings;\n"
+            "found in:\n    {}".format(
+                "\n    ".join(bad_filters)))
+
+
+# @pytest.mark.slow
+def test_warning_calls_stacklevels(warning_calls):
+    bad_filters, bad_stacklevels = warning_calls
+
+    msg = ""
+
+    if bad_filters:
+        msg += ("warning ignore filter should not be used, instead, use\n"
+                "scipy._lib._numpy_compat.suppress_warnings (in tests only);\n"
+                "found in:\n    {}".format("\n    ".join(bad_filters)))
+        msg += "\n\n"
+
+    # tifffile still doesn't use stacklevels everywhere, but we plan to stop
+    # versioning it soon anyway.
+    bad_stacklevels = [item
+                       for item in bad_stacklevels
+                       if 'tifffile.py' not in item]
+
+    if bad_stacklevels:
+        msg += ("warnings should have an appropriate stacklevel:\n"
+                "    {}".format("\n    ".join(bad_stacklevels)))
+
+    if msg:
+        raise AssertionError(msg)
+
+
+
+"""
+Copyright (c) 2001, 2002 Enthought, Inc.
```

Comment:
We distribute all license text in the main LICENSE.txt now.

Reply:
Do you want to remove it from this file as well? I find it useful to keep the license in the file for these single "borrowed" files from libraries.

Reply:
Either way is fine, as long as it is listed in the main license file as well.

Reply:
Done.